boot.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/deis/registry-token-refresher/pkg"
+	"github.com/deis/registry-token-refresher/pkg/credentials"
+	"k8s.io/kubernetes/pkg/api"
+	kcl "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+var appList []string
```

Comment:
exactly my point. unless there's a lock or a single goroutine manages this slice's state, you have a race. prefer the latter